allennlp/commands/elmo_perplexities.py

```diff
@@ -0,0 +1,333 @@
+        """
+        if any([len(s) == 0 for s in batch]):
+            raise ValueError("empty sentence found!")
+
+        perplexities, mask = self.batch_to_preds(batch)
+        perplexities_np = perplexities.data.cpu().numpy()
+        mask_np = mask.data.cpu().numpy()
+        return [perp_i[:, :len_i] for perp_i, len_i in zip(perplexities_np, mask_np.sum(1))]
+
+    def predict_sentences(self,
+                        sentences: Iterable[List[str]],
+                        batch_size: int = DEFAULT_BATCH_SIZE) -> Iterable[numpy.ndarray]:
+        """
+        Computes the ELMo preddings for a iterable of sentences.
+
+        Parameters
+        ----------
+        sentences : ``Iterable[List[str]]``, required
+            An iterable of tokenized sentences.
+        batch_size : ``int``, required
+            The number of sentences ELMo should process at once.
+
+        Returns
+        -------
+            A list of tensors, each representing the ELMo vectors for the input sentence at the same index.
+        """
+        for batch in lazy_groups_of(iter(sentences), batch_size):
+            yield from self.predict_batch(batch)
+
+    def predict_file(self,
+                   input_file: IO,
+                   output_file_path: str,
+                   batch_size: int = DEFAULT_BATCH_SIZE) -> None:
+        """
+        Computes ELMo perplexities from an input_file where each line contains a sentence tokenized by whitespace.
+        The ELMo perplexities are written out in HDF5 format, where each sentence is saved in a dataset.
+
+        NOTE! the first few perplexities will probably be high because it's
+
+        Parameters
+        ----------
+        input_file : ``IO``, required
+            A file with one tokenized sentence per line.
+        output_file_path : ``str``, required
+            A path to the output hdf5 file.
+        output_format : ``str``, optional, (default = "all")
+            The perplexities to output.
+        batch_size : ``int``, optional, (default = 64)
+            The number of sentences to process in ELMo at one time.
+        """
+        # Tokenizes the sentences.
+        sentences = [line.strip() for line in input_file if line.strip()]
+        split_sentences = [sentence.split() for sentence in sentences]
+        # Uses the sentence as the key.
+        perplexities = zip(sentences, self.predict_sentences(split_sentences, batch_size))
+
+        logger.info("Processing sentences.")
+        with h5py.File(output_file_path, 'w') as fout:
+            for key, embeddings in Tqdm.tqdm(perplexities):
+                fout.create_dataset(
+                    key,
+                    embeddings.shape, dtype='float32',
+                    data=embeddings
+                )
+        input_file.close()
+
+
+def elmo_command(args):
+    elmo_model = ElmoPerplexityModel(args.options_file, args.weight_file, args.cuda_device)
```

Comment:
should have 5 arguments. 
(args.options_file, args.weight_file, args.softmax_file, args.vocab_path, args.cuda_device)

https://github.com/allenai/allennlp/pull/1111/files#diff-ce3ea56cea653aba3170d29e0e71e3e3R129